### DISCUSSION TITLE
/oo/abstract-vs-interface.html

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#56] [Announcements]
# /oo/abstract-vs-interface.html

Java程序员进阶之路，小白的零基础Java教程，认真聊聊 Java抽象类和接口的区别

https://tobebetterjavaer.com/oo/abstract-vs-interface.html

**Comment by ghycn** [comment]
1.接口中不能含有静态代码块以及静态方法
2.也就是说对于抽象类，如果需要添加新的方法，可以直接在抽象类中添加具体的实现，子类可以不进行变更；而对于接口则不行，如果接口进行了变更，则所有实现这个接口的类都必须进行相应的改动。

以上两处现在已经不适用了哦。正如在上一篇介绍接口的文章描述一样，接口也支持静态方法和默认方法
